compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/GenerateClosuresForDefaultValues.java

```diff
@@ -524,6 +545,25 @@ private BLangExpression createClosureForDefaultValue(String closureName, String
         env.enclPkg.functions.add(function);
         env.enclPkg.topLevelNodes.add(function);
         symbol.defaultValues.put(paramName, varSymbol);
+        rewrite(lambdaFunction, lambdaFunction.capturedClosureEnv);
+        return returnStmt.expr;
+    }
+
+    private BLangExpression createClosureForDefaultValueOfRecordField(String closureName, String paramName,
```

Comment:
This is pretty much a duplicate of the preceding method, right? Why can't we extract the common logic out to (a) separate method(s)?